MatrixSDKTests/MXClientInformationServiceUnitTests.swift

```diff
@@ -0,0 +1,165 @@
+//
+// Copyright 2022 The Matrix.org Foundation C.I.C
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import XCTest
+@testable import MatrixSDK
+
+class MXClientInformationServiceUnitTests: XCTestCase {
+
+    func testUpdateData() {
+        MXSDKOptions.sharedInstance().enableNewClientInformationFeature = true
+
+        let mockDeviceId = "some_device_id"
+        let credentials = MXCredentials(homeServer: "", userId: "@userid:example.com", accessToken: "")
+        credentials.deviceId = mockDeviceId
+        guard let session = MXSession(matrixRestClient: MXRestClientStub(credentials: credentials)) else {
+            XCTFail("Failed to setup test conditions")
+            return
+        }
+
+        let service = MXClientInformationService(withSession: session)
+
+        let type = service.accountDataType(for: session)
+
+        // no client info before
+        let clientInfo = session.accountData.accountData(forEventType: type)
+        XCTAssertNil(clientInfo)
+
+        service.updateData()
+
+        // must be set after updateData
+        let updatedInfo1 = session.accountData.accountData(forEventType: type)
+        XCTAssertNotNil(updatedInfo1)
```

Comment:
Should the values of `updatedInfo1` be tested as well, not just not nil?

Reply:
It's a bit hard to test the values because `name` and `version` properties should come from the bundle. Now it's like:
`name: xctest iOS, version: 13.4`. Maybe name is standard but you need to read Xcode version to fully test it.

Reply:
Ok. If you want, you can create a protocol something like `ClientInformationProvidable` and inject it into `MXClientInformationService`. In the real code, implementation of that providable would use real Bundle, but in the unit test target, you could create mock. It is just an idea if you would like to do it. Up to you

Reply:
Updated